ethicml/data/util.py

```diff
@@ -12,6 +15,29 @@
     "label_specs_to_feature_list",
 ]
 
+_F = TypeVar("_F", bound=Callable[..., Any])
+
+
+def deprecated(func: _F) -> _F:
+    """This is a decorator which can be used to mark functions as deprecated.
+
+    It will result in a warning being emitted when the function is used.
+    """
+
+    @functools.wraps(func)
+    def new_func(*args: Any, **kwargs: Any) -> Callable:
```

Comment:
This return type is wrong actually. This is not returning a Callable but whatever `func` is returning.

Sorry that I spotted it so late!

Reply:
Looking at the mypy docs, this needs tweaking a little so that the decorator does indeed return a callable https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators